articles/i2p-docker-services/BasicStaticeepSite.md

```diff
@@ -0,0 +1,149 @@
+Creating a simple static eepSite using Docker, Alpine, and i2pd
+===============================================================
+
+One of the most common things I hear people say about i2p is that they have
+trouble finding interesting content on the existing eepSites. It's true, to an
+extent, as well. Most of the interesting eepSite content is, in fact, about i2p
+and i2p-related projects. That's not a bad thing, but it's a little unfortunate
+because it's actually very easy to host your own content on the i2p network. In
+this, the first of many guides to hosting web-like services within the i2p
+network using i2pd and Docker.
+
+Prerequisites
+-------------
+
+The tutorial here assumes that you've already got a GNU/Linux machine with
+Docker installed and have at least a passing familiarity with using Docker from
+the terminal. If you need to install Docker, go [here for Debian-based Distros](https://docs.docker.com/install/linux/docker-ce/debian/),
+[here for Ubuntu-based distros](https://docs.docker.com/install/linux/docker-ce/ubuntu/),
+and go [here for Redhat based distros](https://docs.docker.com/install/linux/docker-ce/fedora/).
+On Alpine Linux, run "apk add docker" as root.
+
+### Create a Docker network off the host
+
+In order to provide static IP addresses for the containers you generate, you'll
+need to create a separate Docker network with an explicitly defined, limited
+subnet.
+
+        docker network create --subnet 172.81.81.0/29 eepsite
+
+### Configuring an i2p router with a single HTTP service tunnel
+
+There are a number of Docker containers that can be used to run i2pd, but for
+the purposes of building services that run on i2pd, I like to make my own. I
+prefer this approach because I like to use the defaults that are used by the
+Debian package, and the easiest way to do that is to just use the Debian
+package. The trade-off, of course, is a somewhat larger container.
+
+#### The Dockerfile
+
+        FROM debian:sid
```

Comment:
Please use three backticks (```) for code formatting, eg:

```Dockerfile
FROM debian:sid
RUN apt-get update && apt-get dist-upgrade -y
RUN apt-get install -y gpg ca-certificates
```

This adds syntax highlighting, which vastly improves readability.

Reply:
I think github missed my edit here, but I switched them all to backticks.